bin/whoami

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/perl
+
+=encoding utf8
+
+=begin metadata
+
+Name: whoami
+Description: display effective user ID
+Author: Oğuz Ersen, oguzersen@protonmail.com
+License: perl
+
+=end metadata
+
+=cut
+
+use strict;
+
+my $user = getpwuid($>);
```

Comment:
Note that getpwuid doesn't work on Windows, so you'll need another way to handle that for that platform.

Reply:
Or, maybe it's more complicated:

https://stackoverflow.com/questions/3526420/how-do-i-get-the-current-user-in-perl-in-a-portable-way

Reply:
I am considering:

    my $user = $^O eq 'MSWin32' ? $ENV{USERNAME} : getpwuid($>);

I am not sure if it will work since I don't have access to a Windows machine right now.

Reply:
Don't settle on anything until you know it will work on Windows. Note the strategies that I noted and what's mentioned in that Stackoverflow post.

Reply:
How about:

```
my $user;

if ($^O eq 'MSWin32') {
    $user = getlogin() || $ENV{USERNAME};
}
else {
    $user = getpwuid($>);
}

print "$user\n";
```

Or should I use `Win32::LoginName`?:

```
my $user;

if ($^O eq 'MSWin32') {
    require Win32;
    $user = Win32::LoginName;
}
else {
    $user = getpwuid($>);
}

print "$user\n";
```

Reply:
I don't know the answers to these. You'll need to try these out in various situations on Windows to see if they work.

But, remember that one of the goals is that we only use core modules. Optionally using a non-core module is fine if the operation doesn't depend on that.